discovery/spec.md

```diff
@@ -778,135 +782,145 @@ upon return of the `202 Accepted` from the original request.
 #### `POST /services`
 
 This MUST update the collection of Services in the Discovery Endpoint to
-contain all Services in the request or fail. Any existing Service identified
-in the request will be completely replaced by the Service definition in the
-request. The body of the request message MUST contain an array of zero or more
-Service entries.
-
-The `id` and `epoch` attributes of the Services MAY be omitted. In such cases
-the discovery endpoint MUST assign an appropriate value to the omitted
-field(s). An appropriate `id` MUST be globally unique. An appropriate `epoch`
-MUST be greater than the existing `epoch` for a service with the same `id`.
-When an `id` is provided any existing Service definition with that
-`id` MUST be completely replaced by the Service definition in the request.
-If an `epoch` is given in the request it MUST be greater than the
-`epoch` of any existing Service of the same `id` or the request MUST fail.
-
-Aside from `id` and `epoch`, unless there is an out of band agreement, all
-mandatory attributes MUST be present in the request, and a Discovery Endpoint
-MUST reject requests that are missing such attributes.
-
-If the Discovery Endpoint is unable to successully add every Service in the
-incoming request then an error MUST be generated and none of the specified
-Services are to be added or updated in the Discovery Endpoint.
-
-The Services in the request MUST be processed in the order in which they are
-given. While any error during the update MUST cause the rollback of any effect
-of the request, the processing of each Service in the request MUST be
-applied as if all previous Services in the request were already completed.
-This means that attribute uniqueness checking, such as on the `name`
-attribute, MUST be done on the new state of all Services as if all previously
-specified Services in the incoming request were already successfully processed.
-
-For example, if a Discovery Endpoint has a Service called `MyService`, it is
-possible to rename that Service to `YourService` and then to add, or update,
-another Service to use `MyService` as its `name`.
-
-Likewise, while it might be less than optimal, it is possible for
-a request to update a Service multiple times if the same `id` is used more
-than once within a request.
-
-The follow responses are defined by this specification:
-- `200 OK` if all the specified Services were processed successfully
-  - The HTTP Response Body MUST include an array of the Service values
-    resulting from the successful processing of the request. The order of
-    those values MUST match the order of the Services in the request. The
-    response MAY include attributes and values which were added by the
-    Discovery Endpoint.
+contain all Services in the request or fail. The body of the request message
+MUST contain an array of zero or more Service entries.
+
+The following rules apply to processing the Services specified in the request:
```

Comment:
Do we want a constraint saying that any given `id` can't occur more than once in a request?

Reply:
I believe 791-792 are intended to add that.

Reply:
Aha - I hadn't spotted that. Thanks!